nio/client/async_client.py

```diff
@@ -1388,9 +1387,14 @@ async def sync_forever(
                     task.cancel()
 
                 raise
+        self._stop_sync_forever = False
 
     def stop_sync_forever(self):
-        """Request that the `sync_forever` loop exits gracefully.
+        """Request that a running `sync_forever` loop exits gracefully or the next call to `sync_forever()` returns
+        immediately without doing any sync.
+
+        If a `sync_forever` loop was running the loop exits and the `_stop_sync_forever` flag is reset.
+        Also each call to `sync_forever()` after the flag was set, resets the flag.
```

Comment:
Given that you removed the first `self._stop_sync_forever = False` towards the beginning, is this true anymore?
```suggestion
        Also each call to `sync_forever()` after the flag was set, resets the flag.
```

Reply:
added flag reset to the try block too.
now it is true ;)